src/sentry/auth/helper.py

```diff
@@ -789,9 +790,19 @@ def _finish_setup_pipeline(self, identity):
         # since only SSO or require 2FA can be enabled
         self.disable_2fa_required()
 
-        self.auth_provider = AuthProvider.objects.create(
-            organization=self.organization, provider=self.provider.key, config=config
-        )
+        with transaction.atomic():
```

Comment:
I don't think this is the best approach as it doesn't dual-write to both tables on update.